### PR TITLE
Rename daml2ts to daml2js

### PR DESCRIPTION
**Pull request by hurryabit**
This is pretty much a search and replace over the whole repo, including
file names.

This does not yet add the `daml codegen js` command. We'll do that
in a separate PR.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

---
This change is [Reviewable](https://reviewable.io/reviews/digital-asset/daml/5394)

### CODE REVIEW

File: daml-assistant/daml-helper/src/DA/Daml/Helper/Run.hs [hunk 1]
```diff
@@ -325,11 +325,11 @@ runJar jarPath mbLogbackPath remainingArgs = do
...
-runDaml2ts :: [String] -> IO ()
-runDaml2ts remainingArgs = do
-    daml2ts <- fmap (</> "daml2ts" </> "daml2ts") getSdkPath
-    withProcessWait_' (proc daml2ts remainingArgs) (const $ pure ()) `catchIO`
-      (\e -> hPutStrLn stderr "Failed to invoke daml2ts." *> throwIO e)
+runDaml2js :: [String] -> IO ()
+runDaml2js remainingArgs = do
+    daml2js <- fmap (</> "daml2js" </> "daml2js") getSdkPath
+    withProcessWait_' (proc daml2js remainingArgs) (const $ pure ()) `catchIO`
+      (\e -> hPutStrLn stderr "Failed to invoke daml2js." *> throwIO e)
```
Comment (cocreature) [inline daml-assistant/daml-helper/src/DA/Daml/Helper/Run.hs:332]: Shouldn’t this say `daml codegen js`. The executable invoked by that seems like an implementation detail.

File: docs/source/daml2js/index.rst [hunk 2]
```diff
@@ -1,25 +1,25 @@
 .. Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
 .. SPDX-License-Identifier: Apache-2.0
 
-``daml2ts`` - TypeScript code generator for DAML
+``daml2js`` - TypeScript code generator for DAML
```
Comment (cocreature) [inline docs/source/daml2js/index.rst:4]: Somewhat similar question.  I think it’s slightly confusing to have this called `daml2js` while the command is called `daml codegen js`.

**Comment by hurryabit** [comment]
I agree regarding the docs. I'll have a walk through the docs today and look at all occurrences of `daml2js` and see where replacing them with `daml codegen js` is the right thing.

**Comment by nemanja-da** [comment]
@hurryabit @robin-da Note that we need to update the architecture diagrams then as well as they're pointing to daml2ts at the moment.

**Pull request merged**